net/httpclient.go

```diff
@@ -21,14 +21,66 @@ const (
 	defaultRefreshInterval = 5 * time.Minute
 )
 
+// Lookuper is an indirection and used to map calls to
+// secrets.SecretsReader's GetSecret to the right parameter.
+type Lookuper interface {
```

Comment:
I can see how the Lookuper solves the previous limitation caused by using the 'secretsPath' field. But we always use now the req.URL as the input for the lookup, and then whatever the Lookup returns, we use that as the key for the GetSecret. Since both the SecretReader and the SecretLookuper can be implemented by the user code, have you considered to just use the SecretReader and use the req.URL as the key? I would probably try to do it that way, because it appears simpler to me.

Reply:
Maybe the `SecretLookuper.Lookup` has the wrong parameter. I guess we should pass the http.Request instead, such that it can differentiate read and write paths.
For example you have 2 tokens "r" and "w", you could implement POST,PATCH,PUT,DELETE to pass token "w" and rest "r".
What do you think?

Reply:
for these cases, I would rather use two SecretReaders (or lookupers), and would leave to the calling code which one to use. The reason is that the secrets package may be good for other things, too, besides only http requests.